pkg/pillar/cmd/zedrouter/acl.go

```diff
@@ -861,7 +860,60 @@ func aceToRules(aclArgs types.AppNetworkACLArgs, ace types.ACE) (types.IPTablesR
 						" programmed due to ACL ID allocation failure",
 						aclRule1.Table, aclRule1.Chain, aclRule1.Rule, aclRule1.Action)
 				}
+
+				// Add a hairpin DNAT rule; need upLink IP
+				extIPs, err := types.GetInterfaceAddrs(*ctx.deviceNetworkStatus, upLink)
```

Comment:
There is a chance that the IP address on uplink interface change. Or the current used uplink port stops working and eve starts using another uplink port. Should we take care of these cases now?

Reply:
As you said, checkAndReprogramNetworkInstances is doing this for an uplink failover from e.g. eth0 to eth1.
But that only handles MASQUERADE rules and not the DNAT rules for the port map. So I'll file a story to do DNAT in general and as part of that take care of the IP address change.

Reply:
Inside the same function doAppNetworkModifyUnderlayNetwork is called for each app attached to the corresponding network instance. This takes care of updating any acl rules.